### ISSUE TITLE
should not resolve symlinks while assessing "bids"  files metadata

### ISSUE THREAD

**Issue by yarikoptic** [pauldmccarthy/fsleyes#68]
spotted on a datalad'ified dataset:

```shell
(git-annex)lena:~/proj/a2cps/data/uic/bids[master]sub-travhuman/func
$> fsleyes sub-travhuman_task-cuff_run-01_bold.nii.gz ../fmap/sub-travhuman_acq-GE_epi.nii.gz 

 WARNING             image.py 1183: __init__        - Failed to load metadata for /home/yoh/proj/a2cps/data/uic/bids/sub-travhuman/func/sub-travhuman_task-cuff_run-01_bold.nii.gz: Does not look like a BIDS file: /home/yoh/proj/a2cps/data/uic/bids/.git/annex/objects/mM/XF/MD5E-s236035062--244606ce4f96b1cc998a3f7a2b1f14e2.nii.gz/MD5E-s236035062--244606ce4f96b1cc998a3f7a2b1f14e2.nii.gz
 WARNING             image.py 1183: __init__        - Failed to load metadata for /home/yoh/proj/a2cps/data/uic/bids/sub-travhuman/fmap/sub-travhuman_acq-GE_epi.nii.gz: Does not look like a BIDS file: /home/yoh/proj/a2cps/data/uic/bids/.git/annex/objects/q0/38/MD5E-s4854971--9ef6a5ca5d2405cfeeee7199309f1128.nii.gz/MD5E-s4854971--9ef6a5ca5d2405cfeeee7199309f1128.nii.gz
^Z
[1]  + 2639110 suspended  fsleyes sub-travhuman_task-cuff_run-01_bold.nii.gz 

$> bg
[1]  + 2639110 continued  fsleyes sub-travhuman_task-cuff_run-01_bold.nii.gz 

$> fsleyes --version
fsleyes/FSLeyes version 0.34.2
```

I guess due to  the use of `realpath` instead of `abspath` somewhere:

```
$> python -c 'import os.path; print(os.path.realpath("sub-travhuman_task-cuff_run-01_bold.nii.gz"))'   
/home/yoh/proj/a2cps/data/uic/bids/.git/annex/objects/mM/XF/MD5E-s236035062--244606ce4f96b1cc998a3f7a2b1f14e2.nii.gz/MD5E-s236035062--244606ce4f96b1cc998a3f7a2b1f14e2.nii.gz

$> python -c 'import os.path; print(os.path.abspath("sub-travhuman_task-cuff_run-01_bold.nii.gz"))' 
/home/yoh/proj/a2cps/data/uic/bids/sub-travhuman/func/sub-travhuman_task-cuff_run-01_bold.nii.gz
```

**Comment by pauldmccarthy** [comment]
Aah, so I guess the name of the symlink should be used, rather than the name of the target file ..

**Comment by yarikoptic** [comment]
> Aah, so I guess the name of the symlink should be used, rather than the name of the target file ..

yes.  But also why not just to use a "full path" for that file without resolving it to some other location/filename?

**Comment by pauldmccarthy** [comment]
I don't know why I had a call to `os.path.realpath` in the [BIDS loader](https://github.com/pauldmccarthy/fslpy/blob/master/fsl/utils/bids.py#L192) - it was a part of the first commit, so there is no explanation for it. I can't see the harm in removing it though, so will do that - I am aiming to have new versions of fsleyes/fslpy out within the next couple of weeks.

**Comment by yarikoptic** [comment]
if you could test on some datalad dataset (http://datasets.datalad.org/ has plenty, and all openneuro datasets are there as well) -- would be great!  e.g. `datalad install -g -J4 ///openneuro/ds000001` should give you the dataset with all data downloaded (you could also go without `-g` and just `datalad get` files of interest ;))

**Comment by pauldmccarthy** [comment]
Cool, I've removed the `realpath` call, added a unit test, and played with a datalad/openneuro dataset - all seems fine without it

**Comment by yarikoptic** [comment]
Great -- thank you!

**Comment by pauldmccarthy** [comment]
This should be fixed as of the latest version of fslpy (3.6.0) - let me know if you run into any more problems
